### PR TITLE
feat: adding queue between add and cohort distribution 

### PR DESCRIPTION
**Pull request by SamRobinson75684**
## Type of changes

- [ ] Refactoring (non-breaking change)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would change existing functionality)
- [ ] Bug fix (non-breaking change which fixes an issue)

## Checklist

- [ ] I am familiar with the [contributing guidelines](../docs/CONTRIBUTING.md)
- [ ] I have followed the code style of the project
- [ ] I have added tests to cover my changes
- [ ] I have updated the documentation accordingly
- [ ] This PR is a result of pair or mob programming

---

## Sensitive Information Declaration

To ensure the utmost confidentiality and protect your and others privacy, we kindly ask you to NOT including [PII (Personal Identifiable Information) / PID (Personal Identifiable Data)](https://digital.nhs.uk/data-and-information/keeping-data-safe-and-benefitting-the-public) or any other sensitive data in this PR (Pull Request) and the codebase changes. We will remove any PR that do contain any sensitive information. We really appreciate your cooperation in this matter.

- [ ] I confirm that neither PII/PID nor sensitive data are included in this PR and the codebase changes.

### CODE REVIEW

**Review by Warren-Pitterson — APPROVED** [review]

File: application/CohortManager/src/Functions/CohortDistributionServices/CreateCohortDistribution/CreateCohortDistribution.cs [hunk 1]
```diff
@@ -11,6 +11,7 @@ namespace NHS.CohortManager.CohortDistributionService;
 using Model;
 using Model.Enums;
 using Data.Database;
+using System.Security.Cryptography.X509Certificates;
```
Comment (MWClayson-NHS) [inline application/CohortManager/src/Functions/CohortDistributionServices/CreateCohortDistribution/CreateCohortDistribution.cs:14]: I dont think this is used

File: application/CohortManager/src/Functions/CohortDistributionServices/CreateCohortDistribution/CreateCohortDistribution.cs [hunk 2]
```diff
@@ -21,109 +22,118 @@ public class CreateCohortDistribution
...
-            await _exceptionHandler.CreateSystemExceptionLogFromNhsNumber(ex, "", "", "", "N/A");
 
-            return req.CreateResponse(HttpStatusCode.BadRequest);
-        }
 
-        if (string.IsNullOrEmpty(requestBody.ScreeningService) || string.IsNullOrEmpty(requestBody.NhsNumber))
+        if (string.IsNullOrWhiteSpace(basicParticipantCsvRecord.ScreeningService) || string.IsNullOrWhiteSpace(basicParticipantCsvRecord.NhsNumber))
         {
-            string logMessage = $"One or more of the required parameters is missing. NhsNumber: {requestBody.NhsNumber} ScreeningService: {requestBody.ScreeningService}";
+            string logMessage = $"One or more of the required parameters is missing. NhsNumber: {basicParticipantCsvRecord.NhsNumber} ScreeningService: {basicParticipantCsvRecord.ScreeningService}";
```
Comment (MWClayson-NHS) [inline application/CohortManager/src/Functions/CohortDistributionServices/CreateCohortDistribution/CreateCohortDistribution.cs:48]: Please don't log the NHS number

File: application/CohortManager/src/Functions/CohortDistributionServices/CreateCohortDistribution/CreateCohortDistribution.cs [hunk 3]
```diff
@@ -21,109 +22,118 @@ public class CreateCohortDistribution
...
             _logger.LogError("One of the functions failed.\nMessage: {Message}\nStack Trace: {StackTrace}", ex.Message, ex.StackTrace);
-            await _exceptionHandler.CreateSystemExceptionLogFromNhsNumber(ex, requestBody.NhsNumber, requestBody.FileName, "", requestBody.ErrorRecord ?? "N/A");
-            return _createResponse.CreateHttpResponse(HttpStatusCode.BadRequest, req);
+            await _exceptionHandler.CreateSystemExceptionLogFromNhsNumber(ex, basicParticipantCsvRecord.NhsNumber, basicParticipantCsvRecord.FileName, "", JsonSerializer.Serialize(basicParticipantCsvRecord.ErrorRecord) ?? "N/A");
+            throw;
         }
     }
 
-    private async Task<HttpResponseData> HandleErrorResponseIfNull(object obj, HttpRequestData req)
+    private async Task HandleErrorResponseIfNullAsync(string errorMessage, CohortDistributionParticipant cohortDistributionParticipant, string fileName)
```
Comment (MWClayson-NHS) [inline application/CohortManager/src/Functions/CohortDistributionServices/CreateCohortDistribution/CreateCohortDistribution.cs:126]: Can you rename the function name, the IfNull seems to be a red herring

File: application/CohortManager/src/Functions/CohortDistributionServices/CreateCohortDistribution/CreateCohortDistribution.cs [hunk 4]
```diff
@@ -21,109 +22,118 @@ public class CreateCohortDistribution
...
+                    await HandleErrorResponseIfNullAsync("the transformed participant returned null from the transform participant function", transformedParticipant, basicParticipantCsvRecord.FileName);
+                    return;
                 }
-                return _createResponse.CreateHttpResponse(HttpStatusCode.OK, req);
+                _logger.LogInformation("participant has been successfully put on the cohort distribution table");
+                return;
             }
-            _logger.LogInformation("Validation error: A rule triggered a fatal error, preventing the cohort distribution record with NHS number {NhsNumber} from being added to the database", participantData.NhsNumber);
-            return _createResponse.CreateHttpResponse(HttpStatusCode.Created, req);
+            var errorMessage = $"Validation error: A rule triggered a fatal error, preventing the cohort distribution record with NHS number {participantData.NhsNumber} from being added to the database";
```
Comment (MWClayson-NHS) [inline application/CohortManager/src/Functions/CohortDistributionServices/CreateCohortDistribution/CreateCohortDistribution.cs:112]: NHS Number

**Review by Warren-Pitterson — APPROVED** [review]

**Review by MWClayson-NHS — APPROVED** [review]

**Pull request merged**